aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -23,19 +17,22 @@ static inline int64_t round_down(int64_t a, int64_t m) {
   return a - (a % m);
 }
 
-template<typename F>
-static void parallel_for(int64_t end, int64_t step, bool parallelize, F func) {
+template <typename F>
+static void _parallel_for(int64_t size, int64_t step, bool parallelize, F func) {
   if (parallelize) {
-    tbb::parallel_for<int64_t>(0, end, step, func);
+    parallel_for(0, size / step, 1, [func, step](int64_t begin, int64_t end) {
+      int64_t k = begin * step;
+      for (int64_t i = begin; i < end; i++, k += step) {
+        func(k);
+      }
+    });
   } else {
-    for (int64_t i = 0; i != end; i += step) {
+    for (int64_t i = 0; i != size; i += step) {
       func(i);
```

Comment:
This function is really weird. Func is applied both in a blocked fashion (and without any epilogue), or in a scalar way? What is going on in here?!